pkg/util/file.go

```diff
@@ -13,3 +17,22 @@ func SetFileModeForPath(name string, mode os.FileMode) error {
 func SetFileModeForFile(file *os.File, mode os.FileMode) error {
 	return file.Chmod(mode)
 }
+
+// ReadFile reads from a file
+func ReadFile(path string) (string, error) {
```

Comment:
I know that this is an existing function that you moved, but I don't like that this infinitely retries on nonexistent files. It should either have a timeout, or accept a context, or both. The function name and docstring should also indicate that it retries at intervals if the file doesn't exist.

For an example of how this might be problematic - if the user points k3s at a nonexistent vpn token file, k3s will just hang forever on startup, when in reality we probably just want to fail immediately if the file doesn't exist. This function makes more sense for its existing use in the codebase, where it's generally used to get content from a file that is dropped by code that is running in parallel to the goroutine trying to read it.

Take a look at `wait.PollImmediateWithContext` elsewhere in the codebase, that's probably what I'd use here.

Reply:
Derek had the same comment, I added a timeout after 4 minutes